packages/session/src/lib/session.ts

```diff
@@ -0,0 +1,283 @@
+import type { ParseOptions, SerializeOptions } from 'cookie'
+import type { CookieOptions } from '@remix-run/cookie'
+import { Cookie } from '@remix-run/cookie'
+
+import { warnOnce } from './warnings.ts'
+
+/**
+ * An object of name/value pairs to be used in the session.
+ */
+export interface SessionData {
+  [name: string]: unknown
+}
+
+/**
+ * Session persists data across HTTP requests.
+ *
+ * Note: This class is typically not invoked directly by application code.
+ * Instead, use a `SessionStorage` object's `getSession` method.
+ */
+export class Session<Data = SessionData, FlashData = Data> {
+  #id: string
+  #map: Map<keyof Data | FlashDataKey<keyof FlashData & string>, unknown>
+  #status: 'new' | 'clean' | 'dirty' | 'destroyed'
```

Comment:
The `new` status feels redundant to me. `clean`, `dirty`, and `destroyed` make sense. But `new` feels the same as `clean` to me in the sense that in neither case does it need to be saved.

Reply:
yeah the new only commit session on modifications alleviated the original use for `new` - removed